### PR TITLE
Filestore optim v2

### PR DESCRIPTION
**Pull request by regit**
Update of  #2735.

It address the remarks and adds a patch that introduces 2 counters to be able to follow how the new system is behaving.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/277
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/60

### CODE REVIEW

File: src/log-filestore.c [hunk 1]
```diff
@@ -64,12 +64,23 @@
 
 static char g_logfile_base_dir[PATH_MAX] = "/tmp";
 
+SC_ATOMIC_DECLARE(uint32_t, filestore_open_file_cnt);  /**< Atomic counter, to link relative event */
```
Comment (inliniac) [inline src/log-filestore.c]: I have no idea what this comment tries to say with 'to link relative event'

Comment (regit) [inline src/log-filestore.c]: Me neither, looks like a copy/paste without removal ;)

File: src/log-filestore.c [hunk 2]
```diff
@@ -484,6 +526,12 @@ static OutputCtx *LogFilestoreLogInitCtx(ConfNode *conf)
         SCLogInfo("forcing magic lookup for stored files");
     }
 
+    const char *write_meta = ConfNodeLookupChildValue(conf, "write-meta");
+    if (write_meta != NULL && !ConfValIsTrue(write_meta)) {
+        FileWriteMetaDisable();
+        SCLogInfo("Will not write meta file");
```
Comment (inliniac) [inline src/log-filestore.c]: this is not a great message in the default output mode. When function names are not printed there is no way for the user to know this is about filestore

Minor nitpick "files" instead "file"

File: src/log-filestore.c [hunk 3]
```diff
@@ -502,9 +550,31 @@ static OutputCtx *LogFilestoreLogInitCtx(ConfNode *conf)
         }
     }
 
+    const char *file_count_str = ConfNodeLookupChildValue(conf, "max-open-files");
+    if (file_count_str != NULL && strcmp(file_count_str, "no")) {
```
Comment (inliniac) [inline src/log-filestore.c]: is ConfValIsTrue or ConfValIsFalse

File: src/util-file.c [hunk 4]
```diff
@@ -149,6 +155,25 @@ void FileForceTrackingEnable(void)
     g_file_force_tracking = 1;
 }
 
+void FileWriteMetaDisable(void)
```
Comment (inliniac) [inline src/util-file.c]: again, please move this meta stuff to log-filestore

**Comment by inliniac** [comment]
Please add a doc update in the next iteration as well, thanks!

File: src/log-filestore.c (hunk 2 above)
Comment (regit) [inline src/log-filestore.c]: Fixed.

File: src/log-filestore.c (hunk 3 above)
Comment (regit) [inline src/log-filestore.c]: Fixed

File: src/util-file.c (hunk 4 above)
Comment (regit) [inline src/util-file.c]: Oups, missread prev comment.

**Pull request closed without merging**